app/models/shipit/deploy_spec/yarn_discovery.rb

```diff
@@ -0,0 +1,49 @@
+module Shipit
+  class DeploySpec
+    module YarnDiscovery
+      def discover_dependencies_steps
+        discover_yarn || super
+      end
+
+      def discover_yarn
+        yarn_install if yarn?
+      end
+
+      def yarn_install
+        [%(yarn install --no-progress)]
+      end
+
+      def discover_review_checklist
+        discover_yarn_checklist || super
+      end
+
+      def discover_yarn_checklist
+        [%(<strong>Don't forget version and tag before publishing!</strong> You can do this with:<br/>
+          yarn version --new-version <strong>&lt;major|minor|patch&gt;</strong> && git push --tags</pre>)] if yarn?
+      end
+
+      def yarn?
+        yarn_lock.exist? && public?
```

Comment:
Yarn and Npm are very similar tools, and both work against the `npm.org` service. It wouldn't seem wrong to handle boths in the `NpmDiscovery` module